resources/sql/queries.sql

```diff
@@ -46,3 +46,60 @@ DELETE FROM services WHERE id = :id;
 -- :name delete-all-services! :! :n
 -- :doc deletes all service records
 DELETE FROM services;
+
+
+-- :name create-layout-table
+-- :command :execute
+-- :result :raw
+-- :doc Create widget layout table
+create table layout (
+  id                   VARCHAR(20) PRIMARY KEY,
+  username              VARCHAR(30),
+  name                  VARCHAR(30),
+  ret_types             BLOB(30),
+  basis                 VARCHAR(30),
+  data_source           VARCHAR(30),
+  type                  VARCHAR(30),
+  icon                  VARCHAR(30),
+  label                 VARCHAR(30),
+  data_grid             BLOB(30),
+  options               BLOB(300));
+
+
+
+-- :name drop-layout-table :!
+-- :doc Drop layout table if exists
+drop table if exists layout;
+
+
+-- :name create-layout! :! :n
+-- :doc creates a new layout record
+INSERT INTO layout
+    (id, username, name, ret_types, basis, data_source, type, icon,
+        label, data_grid, options)
+    VALUES (:id, :username, :name, :ret_types, :basis, :data_source, :type, :icon,
+            :label, :data_grid, :options);
+
+
+-- :name save-layout! :! :n
+-- :doc creates multiple new layout record, ordered by keys
+INSERT INTO layout
+    (basis, data_grid, data_source, icon, id, label, name, options,
+        ret_types, type, username)
+VALUES :tuple*:layout
+ON CONFLICT (id) DO UPDATE SET data_grid=excluded.data_grid;
```

Comment:
I didn't know you could do this

Reply:
Thanks!  Just realized there will be a bug with updating the :options tag when the user changes the widget banner color/text color (whoops).  I think I can just add another field to the update set, will implement and test

https://sqlite.org/lang_UPSERT.html

Reply:
good catch

Reply:
updated the sql insert which works fine, but only works when a new widget has been added after updating the banner color.  Because the :update-color event thats now in basic widget only has the info of the one widget being updated, i need to figure out how to dispatch the update-layout event, but it needs the whole widget list from app-db which i dont think is available anywhere in the basic-widget ns.  Other option is to write a new route and post handler for updating a single widget (sql insert already exists), but wanted to get your opinion if theres an easy way to access the full widget list in basic-widget somehow as that would be simpler and add less clutter

Reply:
I'd recommend writing a new update-layout call that updates the layout of just one widget. That way basic-widget stay focused on just one widget, and since changing colors/title text (current sprint) affects only 1 widget we keep from extending the scope of responsibility.